### ISSUE TITLE
[Broken] Fields Interface

### ISSUE THREAD

**Issue by shealavington** [directus/directus#2386]
On the latest suite & the demo the files and file interfaces are not saving the relation to `directus_files.id`

Using the `files` interface will insert NULL into the mapping table where the file ID should be, and the `file` interface will throw 'internal server error'.

I'm unsure if this is app or api, I think it's the app providing the API with an object which fails, where it should only be and ID passed through.

https://www.loom.com/share/47487a7fcb084f2ca7f813ef6d4d1838

**Comment by benhaynes** [comment]
@rijkvanzanten @bjgajjar — App or API?

**Comment by rijkvanzanten** [comment]
I don't think anything changed in the single file interface in this release 🤔 @bjgajjar any changes in the API that could've caused this?

**Comment by shealavington** [comment]
From my _quick_ debugging before making this issue

The app is trying to send the entire item object `{image:{id:13,title:'thing',type:'image/jpg'}}`
However when looking at how the M2O works, the M2O only sends the ID. `{image:13}`

Following the above and making a post request using `{image:13}` will work without issue, however, sending the object ends up with `internal server error`, so I would believe it's app-based.

Perhaps it's failing to select the ID properly, therefore trying to submit the entire object instead of only the Primary-Key's value. I did at one point see an error in the console for can't select ID of null in a related collection.

I don't know if that helps :man_shrugging:

**Comment by dev7ch** [comment]
from a quick overflew about this issue it sou nds a bit like:
- a field validation could be misconfigured, which returns wrong parsing
- the schema type (field type in app) of the field is changed somewhere wrongly (if so probably be somewhere in app) 
... also not sure if it can help ;) ... remimnds me a bit an errors i got while delaing with diffrent field types and schemas

**Comment by shealavington** [comment]
No defaults have been changed, also, there's nothing to change that could cause it to break, as `Field Type` can only be `File` and the `MySQL Datatype` can only be `int`. _(EG: 'image' can only be '13')_

Thanks for the insight though, I did have a look to be sure :)

**Comment by rijkvanzanten** [comment]
@shealavington you're right, but it has historically always done that, so I'm wondering if something under the hood changed in the API that's now causing issues 🤷‍♂

**Comment by shealavington** [comment]
Perhaps you're right with it being an API change as I did try older app versions, which didn't fix it. Though I find that odd as the relational interfaces only send an ID whereas the file/files interfaces are sending the entire object. :joy:

**Comment by bjgajjar** [comment]
@rijkvanzanten @shealavington  Agree! It is the issue of API. I'll resolve it and send the PR ASAP.
